### ISSUE TITLE
data file name

### ISSUE THREAD

**Issue by wembrey** [ipzero209/Tagmon#1]
If you're using this again you may want to edit line 83 to create the file as data.db, otherwise the script never finds the file because it's missing the ".db" extension.

d = shelve.open('data')
to
d = shelve.open('data.db')

**Comment by ipzero209** [comment]
LIne 83 is creating the shelve object. When the object is created, the .db extension is added automatically.  Line 53 actually opens the shelve object, and refers to 'data.db'. 

For example, the following code creates a shelve object and writes a value:


import shelve

d = shelve.open('data.db')
d['new_key'] = "This is a new key"


The result of this is that a file named 'data.db.db' is created:

drwxr-xr-x   4 cstancill  PALOALTONETWORK\Domain Users   128B Mar 15 07:00 .
drwx------+ 63 cstancill  PALOALTONETWORK\Domain Users   2.0K Mar 15 07:00 ..
-rw-r--r--   1 cstancill  PALOALTONETWORK\Domain Users    16K Mar 15 07:00 data.db.db
-rwxr-xr-x@  1 cstancill  PALOALTONETWORK\Domain Users    96B Mar 15 07:00 sh_test.py

**Comment by wembrey** [comment]
Hi,

When I ran it, it didn’t create the extension. It ran fine after the mod. 

Can you offer advice on running it as a background task or service?

Cheers 

Will

**Comment by ipzero209** [comment]
I did a little digging - were you running it on Windows? It seems the behavior is different depending on platform (and I didn't have cross platform in mind when I wrote it). If there's interest I'll re-write using pickle and include a setup to run it as a service. 

If you want to use this as a service there are a couple of options:

1. Schedule as a cron job and schedule @reboot.
2. Create an entry in /etc/init.d.


Number 1 is likely the easiest. The list of monitored IPs will survive reboot, so there shouldn't be much of an issue.
